cockatrice/src/window_main.cpp

```diff
@@ -421,6 +427,36 @@ MainWindow::~MainWindow()
     clientThread->wait();
 }
 
+void MainWindow::createTrayIcon() {
+    QMenu *trayIconMenu = new QMenu();
+    trayIconMenu->addAction(minimizeAction);
+    trayIconMenu->addAction(maximizeAction);
+    trayIconMenu->addAction(restoreAction);
+    trayIconMenu->addSeparator();
+    trayIconMenu->addAction(closeAction);
+    
+    trayIcon = new QSystemTrayIcon(this);
+    trayIcon->setContextMenu(trayIconMenu);
+    trayIcon->setIcon(QIcon(":/resources/appicon.svg"));
+    trayIcon->show();
```

Comment:
I changed it to `QMenu *trayIconMenu = new QMenu(this);`.
Also, widgets are deleted when the parent is deleted.